lib/sites/walmart.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const siteUtils = require('../site-utils');
+const logger = require('../logger')();
+
+class WalmartSite {
+  constructor(uri) {
+    if (!WalmartSite.isSite(uri)) {
+      throw new Error('invalid uri for Walmart: %s', uri);
+    }
+
+    this._uri = uri;
+  }
+
+  getURIForPageData() {
+    return this._uri;
+  }
+
+  isJSON() {
+    return false;
+  }
+
+  findPriceOnPage($) {
+    // find the price on the page
+    const priceString = $('*[itemprop="price"]').text();
```

Comment:
e2e test cases is where this happens? If yes, then let me look at it and see what I can do!